javaparser-core/src/main/java/com/github/javaparser/printer/PrettyPrinterConfiguration.java

```diff
@@ -42,4 +45,20 @@ public PrettyPrinterConfiguration setPrintComments(boolean printComments) {
         this.printComments = printComments;
         return this;
     }
+
+    public PrettyPrinterConfiguration setVisitorClass(Class<? extends PrettyPrintVisitor> visitorClass) {
+        this.visitorClass = visitorClass;
+        return this;
+    }
+
+    public PrettyPrintVisitor createVisitor() {
+        try {
+            Constructor<? extends PrettyPrintVisitor> constructor =
```

Comment:
Did you mean using generics as follows or moving createVisitor to an extra factory class? Could you be more specific?
```
public class PrettyPrinterConfiguration<T extends PrettyPrintVisitor> {
    private Class<T> visitorClass = (Class<T>) PrettyPrintVisitor.class;

    public T createVisitor() {
        try {
            Constructor<T> constructor =
                    visitorClass.getConstructor(PrettyPrinterConfiguration.class);
            return constructor.newInstance(this);
        } catch (Exception e) {
            assert false;
            return null;
        }
    }
}
```